src/App.js

```diff
@@ -15,33 +18,112 @@ const App = () => {
+        }),
+        method: 'GET',
+        redirect: 'follow',
+      })
+        .then((response) => response.json())
+        .then(({ auth_token }) => auth_token)
+        .catch((error) => console.error(error));
+
+    const fetchStates = async () => {
+      const authorizationToken = await fetchAuthorizationToken();
+
+      await fetch('https://www.universal-tutorial.com/api/states/United States', {
+        headers: new Headers({
+          Accept: 'application/json',
+          Authorization: `Bearer ${authorizationToken}`,
+          'api-token': 'aJC50l2lisn1j0sD7nfyZsu0npkhf6JW2YA3FcQ1244YPtONQwiiIeU0H-6kdd6kbgY',
+          'user-email': 'abc.121009.210823@gmail.com',
+        }),
+        method: 'GET',
+        redirect: 'follow',
+      })
+        .then((response) => response.json())
+        .then((states) => {
+          const hiddenSelect = abdsSelectRef.current?.querySelector('select');
+          const hiddenSelectOptions = hiddenSelect?.querySelectorAll('option');
+          const hiddenSelectPlaceholder = Array.from(hiddenSelectOptions).filter((option) =>
+            option.classList.contains('placeholder')
+          )[0];
+
+          if (abdsSelectRef.current) abdsSelectRef.current.innerHTML = '';
+
+          abdsSelectRef.current?.appendChild(hiddenSelect);
+
+          if (hiddenSelect) hiddenSelect.innerHTML = '';
+
+          hiddenSelect?.appendChild(hiddenSelectPlaceholder);
+
+          states.forEach(({ state_name: state }) => {
+            const option = document.createElement('option');
+            const abdsOption = document.createElement('abds-select-option');
+
+            abdsOption.innerText = state;
+            abdsOption.setAttribute('value', state);
+
+            option.innerText = state;
+            option.setAttribute('value', state);
+
+            abdsSelectRef.current?.appendChild(abdsOption);
+            hiddenSelect?.appendChild(option);
+          });
+        })
+        .catch((error) => console.error('error', error));
+    };
+
+    fetchStates();
+  };
+
+  const initializeTacoTuesday = () => {
+    /**
+     * If a boolean attribute is set in the markup and that markup
+     * then updates on rerender, the attribute will be true on each
+     * rerender. One way to workaround this is to use useEffect to
+     * set the attribute only one time on the initial render.
+     */
+    if (!tacoTooltipRef.current.open) tacoTooltipRef.current.open = true;
+  };
 
-    if (!tacoTuesday.open) tacoTuesday.open = true;
-  }, []);
+  const toggleTheme = () => {
+    const body = document.querySelector('body');
+
+    ['bg-black', 'text-white'].forEach((className) => {
+      body.classList[darkMode ? 'add' : 'remove'](className);
+    });
+  };
+
+  useEffect(initializeStatesSelect, []);
+
+  useEffect(initializeTacoTuesday, []);
+
+  useEffect(toggleTheme, [darkMode]);
 
   const addItem = () => {
     const item = document.createElement('li');
```

Comment:
I think if we break down the App component into smaller pieces, a lot of this will make more sense and ways to refactor further may reveal themselves.

Reply:
if we refactor, we should make sure to include propTypes and default values for props

Reply:
I refactored, but have not implemented propTypes yet. Good idea though. I'll look into adding them where needed.